yoti_python_sdk/anchor.py

```diff
@@ -1,141 +1,178 @@
+                anchor_type = "UNKNOWN"
+                try:
+                    origin_server_certs_list = list(anc.origin_server_certs)
+                    origin_server_certs_item = origin_server_certs_list[0]
+
+                    crypto_cert = crypto.load_certificate(
+                        OpenSSL.crypto.FILETYPE_ASN1, origin_server_certs_item
+                    ).to_cryptography()
+
+                except Exception as exc:
+                    if logging.getLogger().propagate:
+                        logging.warning(
+                            "Error loading anchor certificate, exception: {0} - {1}".format(
+                                type(exc).__name__, exc
+                            )
+                        )
+                    continue
+
+                has_found_anchor = False
+                for i in range(len(crypto_cert.extensions)):
+                    try:
+                        extensions = crypto_cert.extensions[i]
+                        if hasattr(extensions, "oid"):
+                            oid = extensions.oid
+                            if hasattr(oid, "dotted_string"):
+                                if oid.dotted_string == SOURCE_EXTENSION:
+                                    anchor_type = config.ANCHOR_SOURCE
+                                elif oid.dotted_string == VERIFIER_EXTENSION:
+                                    anchor_type = config.ANCHOR_VERIFIER
+
+                                if anchor_type != "UNKNOWN":
+                                    has_found_anchor = True
+                                    parsed_anchors = Anchor.get_values_from_extensions(
+                                        anc,
+                                        anchor_type,
+                                        extensions,
+                                        crypto_cert,
+                                        parsed_anchors,
+                                    )
+
+                    except Exception as exc:
+                        if logging.getLogger().propagate:
+                            logging.warning(
+                                "Error parsing anchor certificate extension, exception: {0} - {1}".format(
+                                    type(exc).__name__, exc
+                                )
+                            )
+                        continue
+
+                if not has_found_anchor:
+                    parsed_anchors = Anchor.get_values_from_extensions(
+                        anc, anchor_type, extensions, crypto_cert, parsed_anchors
+                    )
+
+        return parsed_anchors
+
+    @staticmethod
+    def get_values_from_extensions(
+        anc, anchor_type, extensions, crypto_cert, parsed_anchors
+    ):
+        if hasattr(extensions, "value"):
+            extension_value = ""
+            if hasattr(extensions.value, "value"):
+                extension_value = Anchor.decode_asn1_value(extensions.value.value)
```

Comment:
Will `extensions` always be empty of unknown anchors here or should we skip this when we know it's an unknown _oid_?

Reply:
Yes we don't want to parse the unknown anchors since we won't know the encoding scheme in advance. See SDK-621
for further details. We just want to set the unknown anchor's value to `""`